components/brave_vpn/common/wireguard/wireguard_utils.cc

```diff
@@ -82,6 +143,30 @@ WireguardKeyPair GenerateNewX25519Keypair() {
       EncodeBase64(std::vector<uint8_t>(privkey, privkey + 32)));
 }
 
+// See https://github.com/google/re2/wiki/Syntax for RE2 syntax
+bool IsValidConfig(const wchar_t* config) {
```

Comment:
Don't you think it would be better to have `wcstring_view config` here?

Reply:
We could - the COM interface that is calling this (`IsValidConfig`) is in `browser/brave_vpn/win/brave_vpn_wireguard_service/service/brave_wireguard_manager.cc` (see `EnableVpn`). I didn't want to put any casting logic in the method, just passing the raw config value to the validation.

Reply:
`base::wcstring_view` won't require any casting here though. It will act just like `wchar_t*`.

Reply:
I tried and ran into some casting issues 😦